app/search_builders/sufia/my_search_builder_behavior.rb

```diff
@@ -0,0 +1,14 @@
+# Behavior for the my controller search builder to allow the my controller to only get access to
+# items I have edit access to.
+#
+# Separating as a behavior allows for easier overriding of the gem behavior.
+module Sufia::MySearchBuilderBehavior
+  include Blacklight::Solr::SearchBuilderBehavior
+  include Hydra::AccessControlsEnforcement
+  include Sufia::SearchBuilder
+
+  # show only files with edit permissions in lib/hydra/access_controls_enforcement.rb apply_gated_discovery
+  def discovery_permissions
+    @discovery_permissions = ["edit"]
```

Comment:
@jcoyne I was matching up with how the behavior works in hydra-access-controls. https://github.com/projecthydra/hydra-head/blob/master/hydra-access-controls/lib/hydra/access_controls_enforcement.rb#L81